src/ts/shared/control.ts

```diff
@@ -0,0 +1,396 @@
+   * @param  {String} type optional type of control we want to look up
+   * subtypes of. If not specified will return all types
+   * @return {Array} registered types (or subtypes)
+   */
+  static getRegistered(type = false) {
+    const types = Object.keys(Control.classRegister) as fbControlType[]
+    if (!types.length) {
+      return types
+    }
+    return types.filter(key => {
+      // if type is specified, then we want to return all subtypes
+      // of that type (registered with the key <type>.<subtype>)
+      if (type) {
+        return key.indexOf(`${type}.`) > -1
+      }
+      return key.indexOf('.') == -1
+    })
+  }
+
+  /**
+   * Retrieves an object of types mapped to an array of subtypes.
+   * Only returns types that have subtypes
+   * @return {Object} an object containing {type: array of subtypes}.
+   */
+  static getRegisteredSubtypes(): SubTypeOptions {
+    const types = {}
+    for (const key in Control.classRegister) {
+      if (Control.classRegister.hasOwnProperty(key)) {
+        const [type, subtype] = key.split('.')
+        if (!subtype) {
+          continue
+        }
+        if (!types[type]) {
+          types[type] = []
+        }
+        types[type].push(subtype)
+      }
+    }
+
+    return types
+  }
+
+  /**
+   * Retrieve the class for a specified control type
+   * @param {String} type type of control we are looking up
+   * @param {String} subtype if specified we'll try to find
+   * a class mapped to this subtype. If none found, fall back to the type.
+   * @return {Class} control subclass as defined in the call to register
+   */
+  static getRegisteredClassControl(type: fbControlType, subtype: fbControlSubtype = null): typeof Control {
+    const lookup = subtype ? `${type}.${subtype}` : type
+    const control = Control.classRegister[lookup] || Control.classRegister[type]
+    if (!control) {
+      throw new Error(
+        `Invalid control type. (Type: ${type}, Subtype: ${subtype}). Please ensure you have registered it, and imported it correctly.`,
+      )
+    }
+
+    return control
+  }
+
+  /**
+   * support dynamic loading of custom control classes
+   * @param {Array} controls
+   */
+  static loadCustom(controls) {
+    let controlClasses = []
+    if (controls) {
+      controlClasses = controlClasses.concat(controls)
+    }
+
+    // support for user loaded plugin controls
+    if (window.fbControls) {
+      controlClasses = controlClasses.concat(window.fbControls)
+    }
+
+    // loop through each defined custom control.
+    // expects a function that receives the master control class to inherit from (or optional classRegister to inherit from subclass)
+    // see src/ts/control_plugins/ for an example
+    if (!this.fbControlsLoaded) {
+      for (const loadControl of controlClasses) {
+        //Kevin - is there an example of how do this whole custom Control process via passing into the controls configuration? I didn't find anything on the website
```

Comment:
this part of the docs explains registering a new control plugin: https://formbuilder.online/docs/formBuilder/control-plugins/#creating-a-new-control-plugin